components/project-loader/project-loader.jsx

```diff
@@ -3,14 +3,14 @@ import Service from '../../js/service.js';
 import ProjectList from '../project-list/project-list.jsx';
 
 export default React.createClass({
+  promiseToken: null,
```

Comment:
this is now a prototype-leve property: better to not put it here, and instead explicitly rely on `this.promiseToken` in the code (exploiting the fact that if there none declared yet, `this.promiseToken` is `undefined`, which is falsey)